File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_3MoreExercise/_01MatchTickets.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._3MoreExercise;

import java.util.Scanner;

public class _01MatchTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double budget = Double.parseDouble(scanner.nextLine());
        String ticketsType = scanner.nextLine();
        int numberOfPeople = Integer.parseInt(scanner.nextLine());
        double transportCost = 0.0;
        double tiketPrice = 0.0;
        if (numberOfPeople >= 1 && numberOfPeople <= 4) {
            transportCost = 0.75 * budget;
        } else if (numberOfPeople >= 5 && numberOfPeople <= 9) {
            transportCost = 0.60 * budget;
        } else if (numberOfPeople >= 10 && numberOfPeople <= 24) {
            transportCost = 0.50 * budget;
        } else if (numberOfPeople >= 25 && numberOfPeople <= 49) {
            transportCost = 0.40 * budget;
        } else if (numberOfPeople > 50) {
            transportCost = 0.25 * budget;
        }
        budget = budget - transportCost;
        double totalTicketCost = 0.0;
        if (ticketsType.equals("VIP")) {
            tiketPrice = 499.99;
            totalTicketCost = tiketPrice * numberOfPeople;
        } else if (ticketsType.equals("Normal")) {
            tiketPrice = 249.99;
            totalTicketCost = tiketPrice * numberOfPeople;
        }
        Double diff = Math.abs(totalTicketCost - budget);
        if (budget >= totalTicketCost) {
            System.out.printf("Yes! You have %.2f leva left.", diff);
        } else {
            System.out.printf("Not enough money! You need %.2f leva.", diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_3MoreExercise/_01MatchTickets.js
function matchTickets(budget, category, numberOfPeople) {
    const VIP_TICKET_PRICE = 499.99;
    const NORMAL_TICKET_PRICE = 249.99;

    const PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_1_TO_4 = 0.75;
    const PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_5_TO_9 = 0.60;
    const PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_10_TO_24 = 0.50;
    const PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_25_TO_49 = 0.40;
    const PERCENT_FOR_TRANSPORT_IF_PEOPLE_MORE_THAN_50 = 0.25;

    let moneyLeftForTickets = 0.00;

    if (numberOfPeople <= 4) {
        moneyLeftForTickets = budget - (budget * PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_1_TO_4);
    } else if (numberOfPeople <= 9) {
        moneyLeftForTickets = budget - (budget * PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_5_TO_9);
    } else if (numberOfPeople <= 24) {
        moneyLeftForTickets = budget - (budget * PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_10_TO_24);
    } else if (numberOfPeople <= 49) {
        moneyLeftForTickets = budget - (budget * PERCENT_FOR_TRANSPORT_IF_PEOPLE_FROM_25_TO_49);
    } else if (numberOfPeople > 50) {
        moneyLeftForTickets = budget - (budget * PERCENT_FOR_TRANSPORT_IF_PEOPLE_MORE_THAN_50);
    }

    let totalTicketPrice = 0.00;
    if (category === "VIP") {
        totalTicketPrice = numberOfPeople * VIP_TICKET_PRICE;
    } else if (category === "Normal") {
        totalTicketPrice = numberOfPeople * NORMAL_TICKET_PRICE;
    }

    const diff = Math.abs(moneyLeftForTickets - totalTicketPrice);

    if (moneyLeftForTickets >= totalTicketPrice) {
        console.log(`Yes! You have ${diff.toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${diff.toFixed(2)} leva.`);
    }
}